### ISSUE TITLE
Nested types not cast correctly to sub-class when data loaded via "expand" 

### ISSUE THREAD

**Issue by antonantonovich** [davidmoten/odata-client#438]
I have a schema with Document and property content of base type ContentInfo. I have some nested types for ContentInfo like File, URL, etc.
When I run getDocument("id").get().getContent().get() I have correct sub-class of ContentInfo.
But when I run  getDocument("id").expand("Content").get().getContent().get() I get the instance of base ContentInfo, not the subclass.

I see in EntityRequest class in get method that
```
if (value.isPresent()) {
    String json = Serializer.INSTANCE.serialize(value.get());
    return Serializer.INSTANCE.deserialize(json, cls, contextPath, false);
} else {
    return RequestHelper.get(contextPath, cls, options);
}
```
So if we have preloaded values -- it always will be deserialized to the base class, but if not, then in RequestHelper we have proper deserialization to correct sub-class:
```
Class<? extends T> c = getSubClass(cp, contextPath.context().schemas(), returnCls, response.getText());
```

**Comment by davidmoten** [comment]
Thanks, I'll have a look.

**Comment by davidmoten** [comment]
Yep I don't have test coverage of abstract complex types. Should be easy to implement, the work will be in putting together the unit tests. Probably done in the next couple of days.

**Comment by davidmoten** [comment]
This fix for this is [0.1.95](https://github.com/davidmoten/odata-client/releases/tag/0.1.95) and is on Maven Central now. Please give it a try.
